code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -6047,6 +6050,18 @@
 	custom_metabolism = 0.01
 	dupeable = FALSE
 
+/datum/reagent/ethanol/scientists_serendipity/steve(var/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/D)
+	var/datum/reagents/glassreagents = D.reagents
+	if(glassreagents.get_reagent_amount(SCIENTISTS_SERENDIPITY)<10)
```

Comment:
you should be able to just use 
```suggestion
	if(amount<10)
```
since `src` is the reagent here

Reply:
Needs to check for scientist's serendipity specifically or else you could put scientist's serendipity + something else equaling 10u into the glass and get the research boost.

Reply:
`volume` here will be the amount of serendipity specifically.

Reply:
sweet, done